protocol/signalfx/signalfxforwarder.go

```diff
@@ -213,6 +198,50 @@ func (connector *Forwarder) coreDatapointToProtobuf(point *datapoint.Datapoint)
 	return v
 }
 
+func (connector *Forwarder) coreEventToProtobuf(e *event.Event) *com_signalfx_metrics_protobuf.Event {
+	ts := e.Timestamp.UnixNano() / time.Millisecond.Nanoseconds()
+	et := e.EventType
+	cat := com_signalfx_metrics_protobuf.EventCategory_USER_DEFINED
+	if catIndex, ok := com_signalfx_metrics_protobuf.EventCategory_value[e.Category]; ok {
+		cat = com_signalfx_metrics_protobuf.EventCategory(catIndex)
+	}
+
+	v := &com_signalfx_metrics_protobuf.Event{
+		EventType:  &et,
+		Timestamp:  &ts,
+		Properties: mapToProperties(e.Meta),
+		Dimensions: mapToDimensions(e.Dimensions),
+		Category:   &cat,
+	}
+	return v
+}
+
+func mapToProperties(properties map[string]interface{}) []*com_signalfx_metrics_protobuf.Property {
+	ret := make([]*com_signalfx_metrics_protobuf.Property, 0, len(properties))
+	for k, v := range properties {
+		if k == "" || v == nil {
+			continue
+		}
+		copyOfK := filterSignalfxKey(string([]byte(k)))
+
+		pv := com_signalfx_metrics_protobuf.PropertyValue{}
+		if ival, ok := v.(int64); ok {
+			pv.IntValue = &ival
+		} else if bval, ok := v.(bool); ok {
+			pv.BoolValue = &bval
+		} else if dval, ok := v.(float64); ok {
+			pv.DoubleValue = &dval
+		} else if sval, ok := v.(string); ok {
+			pv.StrValue = &sval
+		}
```

Comment:
I think it's reasonable to `continue` in this situation.  You don't want to send in a PropertyValue that has no Value set.